blaze/server/server.py

```diff
@@ -113,6 +114,32 @@ def authorized(*args, **kwargs):
     return authorized
 
 
+def check_request(f):
```

Comment:
I moved this stuff to a decorator since it was copied in the `compserver` and `addserver` functions.